app.go

```diff
@@ -787,9 +788,16 @@ func (app *App) Stop(ctx context.Context) (err error) {
 // using the Shutdown functionality (see the Shutdowner documentation for details).
 func (app *App) Done() <-chan os.Signal {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, _sigINT, _sigTERM)
 
+	// If shutdown signal has been received already
+	// send it else wait for user to send a termination
+	// signal.
 	app.donesMu.Lock()
+	if app.shutdownSig != nil {
+		c <- app.shutdownSig
+	} else {
```

Comment:
In this case, we'll want to return the channel early.
We don't need to hold onto the channel.
(It doesn't cause any problems,
but we'll never hit that codepath again after shutdown.)

I'd suggest using a `defer unlock` on the mutex above
so that we can return early here.

```
mu.Lock()
defer mu.Unlock()

if sig != nil {
  c <- sig
  return c
}

...
```